test/Table_Tests/src/In_Memory/Table_Conversion_Spec.enso

```diff
@@ -0,0 +1,201 @@
+            expected = Table.new [["aaa", [1, 2]], ["bbb first", ["Mary", Nothing]], ["bbb last", ["Smith", Nothing]], ["bbb age", [23, Nothing]], ["bbb height", [Nothing, 1.9]], ["bbb weight", [Nothing, 70]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" . should_equal expected
+
+        Test.specify "Expands a non-uniform column of JSON objects with extra and ignored fields" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", non_uniform_json], ["ccc", [5, 6]]]
+            expected = Table.new [["aaa", [1, 2]], ["bbb last", ["Smith", Nothing]], ["bbb height", [Nothing, 1.9]], ["bbb foo", [Nothing, Nothing]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" ["last", "height", "foo"] . should_equal expected
+
+        Test.specify "accept vectors/arrays within a column" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", [[1, 2, 3], [4, 5, 6].to_array]]]
+            expected = Table.new [["aaa", [1, 2]], ["bbb Value", [[1, 2, 3], [4, 5, 6].to_array]]]
+            table.expand_column "bbb" . should_equal expected
+
+        Test.specify "will work even if keys are not Text" <|
+            table = Table.new [["a", [1, 2]], ["b", [Map.from_vector [[1, "x"], [2, "y"]], Map.from_vector [[2, "z"], [3, "w"]]]]]
+            expected = Table.new [["a", [1, 2]], ["b 1", ["x", Nothing]], ["b 2", ["y", "z"]], ["b 3", [Nothing, "w"]]]
+            table.expand_column "b" . should_equal expected
+
+            table2 = Table.new [["a", [1, 2]], ["b", [Map.from_vector [[My_Mod_Type.Value 12, "x"], [My_Mod_Type.Value 23, "y"]], Map.from_vector [[My_Mod_Type.Value 32, "z"]]]]]
+            expected2 = Table.new [["a", [1, 2]], ["b x%10=2", ["x", "z"]], ["b x%10=3", ["y", Nothing]]]
+            table2.expand_column "b" . should_equal expected2
+
+        Test.specify "will fail if text representation of keys is not unique" <|
+            k1 = My_Mod_Type.Value 12
+            k2 = My_Mod_Type.Value 32
+            m = Map.from_vector [[k1, "a"], [k2, "b"]]
+            m.at k1 . should_equal "a"
+            m.at k2 . should_equal "b"
+            k1.to_text . should_equal "x%10=2"
+            k2.to_text . should_equal "x%10=2"
+
+            table = Table.new [["a", [1, 2]], ["b", [Map.from_vector [[k1, "x"], [k2, "y"]] , Map.from_vector []]]]
+            r = table.expand_column "b"
+            r.should_fail_with Illegal_Argument
+            r.catch.to_display_text . should_contain "keys are duplicated when converted to text"
+
+        Test.specify "will ??? when there are no fields" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", [Map.from_vector [], Map.from_vector []]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" . print
+
+        Test.specify "will ??? when fields=[]" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", uniform_json], ["ccc", [5, 6]]]
+            table.expand_column "bbb" fields=[] . print
```

Comment:
I'm not sure what the expected behaviour should be in these 2 cases.

Currently it fails with `Illegal_Argument.Error 'Cannot create a table with no columns.'`.

I think if the user specifies `fields=[]` that should definitely be an error - probably `Illegal_Argument.Error 'The fields parameter cannot be empty.'`.

But I'm not sure what to do if all objects are empty. It's a rather rare edge case for sure. The intuitive 'failsafe' is to remove the column and not add anything back - but that would be rather weird IMO. So maybe it should indeed throw an error? `'Unable to discover expected column names, because all input objects had no fields in them. Specify fields explicitly if you need a constant set of expected columns'`? Maybe something like this?

Reply:
I've implemented the error scenarios as suggested in the comment above, but I'm still open to discussion if an alternative should be chosen instead.

Reply:
Agree with `field=[]` throwing and early error above.

Like the second error too - just dropping the column feels wrong.